sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs

```diff
@@ -2105,5 +2107,249 @@ Response<Blobs.Models.BlobDownloadInfo> response
             }
         }
         #endregion Read Data
+
+        #region Download Data
+        /// <summary>
+        /// The <see cref="DownloadTo(Stream, DataLakeRequestConditions, StorageTransferOptions, CancellationToken)"/>
+        /// operation downloads an entire file using parallel requests,
+        /// and writes the content to <paramref name="destination"/>.
+        /// </summary>
+        /// <param name="destination">
+        /// A <see cref="Stream"/> to write the downloaded content to.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add conditions on
+        /// the download of this file.
+        /// </param>
+        /// <param name="transferOptions">
+        /// Optional <see cref="StorageTransferOptions"/> to configure
+        /// parallel transfer behavior.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> describing the operation.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response DownloadTo(
+            Stream destination,
+            DataLakeRequestConditions conditions = default,
+            //IProgress<long> progressHandler = default,
```

Comment:
Is there a plan on bringing the progress handler back?

Reply:
We had an issue for it here
https://github.com/Azure/azure-sdk-for-net/issues/8506

Reply:
We need to implement it for downloads across all services - https://github.com/Azure/azure-sdk-for-net/issues/8506